### PR TITLE
Fix compile errors with IAR

### PR DESCRIPTION
**Pull request by MartinBP**
Fix the following errors when compiling with IAR
Error[Pe546]: transfer of control bypasses initialization of: variable "attributes"
Error[Pe546]: transfer of control bypasses initialization of: variable "expected_length"

Signed-off-by: Martin Boye Petersen <martinboyepetersen@hotmail.com>

## Description

Please write a few sentences describing the overall goals of the pull request's commits.

## Gatekeeper checklist

- [ ] **changelog** provided, or not required
- [ ] **backport** done, or not required
- [ ] **tests** provided, or not required

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

File: library/bignum.c [hunk 1]
```diff
@@ -2045,7 +2048,7 @@ int mbedtls_mpi_random( mbedtls_mpi *X,
         return( ret );
 
     return( mbedtls_mpi_core_random( X->p, min, N->p, X->n, f_rng, p_rng ) );
-}
+        }
```
Comment (tom-cosgrove-arm) [inline library/bignum.c:2051]: this is an unnecessary whitespace change

**Comment by daverodgman** [comment]
I think we now build 100% cleanly with IAR - I can't manage to construct a configuration that fails. So I'm closing this PR, but please re-open if there is anything more we should pick up.
